src/kernel/thread/stack_protect.c

```diff
@@ -0,0 +1,49 @@
+#include <kernel/thread/stack_protect.h>
+
+#include <assert.h>
+#include <util/binalign.h>
+#include <hal/mmu.h>
+#include <mem/vmem/vmem_alloc.h>
+
+static int enabled = 0;
```

Comment:
Is this really required?
Do you want to protect from using your routines before mmu was enabled?